cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_service_manager.go

```diff
@@ -641,8 +688,17 @@ func buildGenericLabels(template *asgTemplate, nodeName string) map[string]strin
 	result[apiv1.LabelTopologyZone] = template.zone
 	result[apiv1.LabelHostname] = nodeName
 
+	r, err := regexp.Compile("k8s.io/cluster-autoscaler/node-template/taint/(.*)")
+	if err != nil {
+		klog.Errorf("Compile regex error when build labels:%v", err)
+		return result
+	}
 	// append custom node labels
 	for key, value := range template.tags {
+		// ignore the key which contains taints info
+		if r.MatchString(key) {
+			continue
+		}
```

Comment:
```suggestion
		// ignore the tag which represents a taint
		if strings.Contains(key, "k8s.io/cluster-autoscaler/node-template/taint/") {
			continue
		}
```
We don't need a regular expression for this case. since `strings.Contains` would be cheaper.